lib/bibtex/names.y

```diff
@@ -134,7 +157,8 @@ require 'strscan'
   def scan(input)
     @src = StringScanner.new(input)
     @brace_level = 0
-    @stack = []
+    @last_and = 0
+    @stack = [[:ANDFL,'#dummy']]
```

Comment:
I think we need dummy variable, but its name can be changed for the sake of legibility of error messages. Perhaps just `^` is more readable. Considering that racc uses `$` for the end of input, assuming user knows the meaning of `^` is reasonable. Another more verbose possibility is `#dummy_for_leading` or so.